Replicator/Pusher+Revs.cc

```diff
@@ -72,6 +74,22 @@ namespace litecore::repl {
             c4err = C4Error::make(LiteCoreDomain, kC4ErrorNotFound);
         }
 
+        // Encrypt any encryptable properties
+        MutableDict encryptedRoot;
+        if (root && MayContainPropertiesToEncrypt(doc->getRevisionBody())) {
+            logVerbose("Encrypting properties in doc '%.*s'", SPLAT(request->docID));
+            encryptedRoot = EncryptDocumentProperties(request->docID, root,
+                                                      _options.propertyEncryptor,
+                                                      _options.callbackContext,
+                                                      &c4err);
+            if (encryptedRoot)
+                root = encryptedRoot;
+            else if (c4err) {
```

Comment:
This means that `EncryptDocumentProperties` is guaranteed to throw an error right?  Is there a case in which `encryptedRoot` is `null` but there is no error?  I think that should also fail the document since no encryption took place.

Reply:
The header answers these questions -- it's marked `noexcept`, and the `@return` line says it will return null if no encryption took place.

No encryption is not an error, since there might not be any actual encrypted properties -- MayContainPropertiesToDecrypt is a very quick check that can return false positives.